client/components/credit-card/stored-card.jsx

```diff
@@ -9,6 +9,27 @@ import classNames from 'classnames';
 import PropTypes from 'prop-types';
 import { localize } from 'i18n-calypso';
 
+export const getCreditCardSummary = ( translate, type, digits ) => {
```

Comment:
🤔 Are we exporting this for a test? It's fine if we are.

No action here needed, but wanted to note that StoredCard isn't a stateless functional component, so if you wanted to, `getCreditCardSummary` could be a class method, and we wouldn't need to pass through `translate` as an argument.

Reply:
Good question – the code I'd adapted exports the function [[#](https://github.com/Automattic/wp-calypso/blob/5aa3e70461afe3527f8d30310e85de2712623f42/client/extensions/woocommerce/woocommerce-services/views/label-settings/label-payment-method.js#L18)], not for a test but to refer to the card as a text string inline:

<img width="505" src="https://user-images.githubusercontent.com/1867547/44609830-4d2e2380-a7c7-11e8-821f-48c170a58569.png">

Would importing and using `translate` directly from the library be preferable to having it passed from above? And in any case, would having this export in a separate file improve organization?

Reply:
> Would importing and using translate directly from the library be preferable to having it passed from above? And in any case, would having this export in a separate file improve organization?

No preference in this case, and I'd only move it into a separate file if we were reusing it elsewhere.

 This stuck out to my 👀 since I'd expect this style for a stateless functional component, like in your [example](https://github.com/Automattic/wp-calypso/blob/5aa3e70461afe3527f8d30310e85de2712623f42/client/extensions/woocommerce/woocommerce-services/views/label-settings/label-payment-method.js#L43) vs tucking away related methods within the class instance.

The alternative would look like:
```jsx
class StoredCard extends Component {

    //...
    getCreditCardSummary() {
        // we can still pass through cardType/digits if you'd like render to sanitize (or some other util to do so).
        const { translate, cardType, digits } = this.props;
        //...
    };

    render() {
        //...
        return (
           <div className={ cardClasses }>
                <span className="credit-card__stored-card-number">
                       { this.getCreditCardSummary() }
          //...
         );
    }
} 
```

I mainly prefer stateless function components when it's a super simple UI piece that's also short. In many cases, components get longer, or require component state or lifecycle methods so we typically end up extending Component.